tests_e2e/tests/lib/agent_test_context.py

```diff
@@ -65,6 +65,7 @@ def __init__(self, working_directory: Path, vm: VirtualMachineClient, ip_address
         super().__init__(working_directory, username, identity_file, ssh_port)
         self.vm: VirtualMachineClient = vm
         self.ip_address: str = ip_address
+        self.private_ip_address: str = self.vm.get_private_ip_address()
```

Comment:
minor: since most tests do not need the private IP address and this involves a network call, consider getting the address on demand